src/lib.rs

```diff
@@ -555,16 +553,23 @@ impl <'a, T> FixedVec<'a, T> where T: 'a + Copy {
     /// vec.push_all(&[1, 2, 3]).unwrap();
     /// {
     ///     let mut iter = vec.iter();
-    ///     assert_eq!(iter.next(), Some(1));
-    ///     assert_eq!(iter.next(), Some(2));
-    ///     assert_eq!(iter.next(), Some(3));
+    ///     assert_eq!(iter.next(), Some(&1));
+    ///     assert_eq!(iter.next(), Some(&2));
+    ///     assert_eq!(iter.next(), Some(&3));
     ///     assert_eq!(iter.next(), None);
     /// }
     /// # }
     /// ```
     #[inline]
     pub fn iter(&'a self) -> Iter<'a, T> {
-        Iter{list: self, idx: 0}
+        let (slice, _) = self.memory.split_at(self.len);
+        slice.iter()
+    }
+
+    #[inline]
+    pub fn iter_mut(&'a mut self) -> IterMut<'a, T> {
```

Comment:
The `'a` lifetimes actually need to be removed here because otherwise the iterator will be tied to the lifetime of the `FixedVec`, and will cause ownership issues if the `FixedVec` is accessed after `iter_mut` has been called. That means that code like the following would fail:

```rust
let mut space = alloc_stack!([u8; 10]);                                                                                                                                                                                          
let mut vec = FixedVec::new(&mut space);                                                                                                                                                                                         
vec.push_all(&[1, 2, 3]).unwrap();                                                                                                                                                                                               
{                                                                                                                                                                                                                                
    let mut iter = vec.iter_mut();                                                                                                                                                                                               
    let mut x = iter.next().unwrap();                                                                                                                                                                                            
    *x = 5;                                                                                                                                                                                                                      
}                                                                                                                                                                                                                                
assert_eq!(vec.as_slice(), &[5, 2, 3]);                                                                                                                                                                                          
```